plutus-metatheory/src/Algorithmic/CEKC.lagda.md

```diff
@@ -95,6 +97,10 @@ BUILTIN verifySignature σ ((_ ,, _ ,, V-con (bytestring k) ,, _) ,, (_ ,, _ ,,
 BUILTIN equalsByteString σ ((_ ,, _ ,, V-con (bytestring b) ,, _) ,, (_ ,, _ ,, V-con (bytestring b') ,, _) ,, tt) = _ ,, con (bool (equals b b')) ,, []
 BUILTIN ifThenElse σ ((_ ,, _ ,, V-con (bool true) ,, _) ,, (_ ,, t ,, _ ,, ρ) ,, _ ,, tt) = _ ,, t ,, ρ
 BUILTIN ifThenElse σ ((_ ,, _ ,, V-con (bool false) ,, _) ,, _ ,, (_ ,, u ,, _ ,, ρ) ,, tt) = _ ,, u ,, ρ
+BUILTIN charToString _ ((_ ,, _ ,, V-con (char c) ,, _) ,, tt) = _ ,, con (string (primStringFromList L.[ c ])) ,, []
+BUILTIN append _ ((_ ,, _ ,, V-con (string s) ,, _) ,, (_ ,, _ ,, V-con (string t) ,, _) ,, tt) =
+  _ ,, con (string (primStringAppend s t)) ,, []
```

Comment:
@jmchapman  @michaelpj This is not very future-proof. We'll probably add general lists to the universe eventually and then `String` will no longer be any special (i.e. it'll be just `[Char]` in PLC as in Haskell), but `String` in Agda is not equal to `List Char`, so we have a mismatch. Probably not a big problem, but looks annoying.